### PR TITLE
Lab 8

### PR DESCRIPTION
**Pull request by borjasotomayor**
I still need to flesh out some parts of the lab, but it should be possible to start testing the instructions on the lab, etc.

### CODE REVIEW

File: website/content/labs/lab8.md [hunk 1]
```diff
@@ -0,0 +1,376 @@
...
+* Commit and push this change with the message "Update to 2.0"
+* It should not break the tests, so Travis CI should succeed, and from the past lab, your app will automatically deploy.
+
+Now, notice that if you navigate to your staging website (CNETID-cs220-lab7.herokuapp.com) you’ll see your change. But if you navigate to your production website (CNETID-cs220-prod.herokuapp.com), you'll still see a placeholder page. Why is this? 
+
+Changes from staging are *not* automatically deployed to production. This is intentional: we often don’t want things to automatically deploy to production, since we might risk breaking the public-facing version of our app. Pushing code that is broken is known as "breaking the build", and you absolutely don't want that to propagate to production (by the way, if you know anyone who has done a software development internship or works in software development, ask them if they've ever "broken the build"; We guarantee you'll hear some entertaning stories) By having a pipeline, we can ensure that users continuously see our production, without mistakes, and we only update it when we’re absolutely sure.
```
Comment (mpcsh) [inline website/content/labs/lab8.md]: "; We guarantee" -> "; we guarantee"

File: website/content/labs/lab8.md [hunk 2]
```diff
@@ -0,0 +1,376 @@
...
+
+An alternative approach is to use _containers_, a sort of lightweight virtual machine that provides a specific software environment we need. Unlike virtual machines, containers are typically used to package a specific piece of software, instead of providing a full OS environment (although this is also possible with containers)
+
+For example, a Redis container would include a Redis server and, by default, running the container would immediately start that Redis server, instead of taking us to a UNIX shell where we have to run the Redis serve ourselves. So, if we wanted to run Redis during a CI build, all we would need to do is download a Redis _container image_, and run that container during the CI build. A popular container manager is Docker, which we will be using in this lab. Make sure to read their [What is a Container](https://www.docker.com/what-container) page, which provides a more in-depth explanation of what a container is, and how it compares to virtual machines.
```
[the remark this answers is not in the document]
Comment (borjasotomayor) [reply website/content/labs/lab8.md]: Fixed in fe72c9b

File: website/content/labs/lab8.md [hunk 3]
```diff
@@ -0,0 +1,376 @@
...
+
+The repository we will be forking is this one: https://github.com/uchicago-cs/cs220-redis-example. Remember that you can fork it by clicking "Fork" in the upper right hand corner.
+
+Once the fork has completed, you can clone it and begin your work. Remember to do this outside of your individual GitLab repository! (i.e., do not run the following command in the same directory that contains your lab1, lab2, etc. directories, or inside the lab8 directory).
+
+    $ git clone https://github.com/[yourusername]/cs220-redis-example.git
+
+Edit your tasks.txt file and include the URL of your forked cs220-redis-example repository on GitHub.
+
+Careful! Remember that your tasks.txt is in your personal repository on GitLab (with all your other lab files). You will only be using that repository to edit your tasks.txt file. The rest of your work will happen on the forked cs220-redis-example repository you just created.
```
[the remark this answers is not in the document]
Comment (borjasotomayor) [reply website/content/labs/lab8.md]: Fixed in fe72c9b

File: website/content/labs/lab8.md [hunk 4]
```diff
@@ -0,0 +1,376 @@
...
+    Cloning into '/tmp/cs220-redis-example/hiredis'...
+    remote: Counting objects: 2909, done.
+    remote: Compressing objects: 100% (13/13), done.
+    remote: Total 2909 (delta 9), reused 19 (delta 9), pack-reused 2887
+    Receiving objects: 100% (2909/2909), 866.09 KiB | 5.16 MiB/s, done.
+    Resolving deltas: 100% (1791/1791), done.
+
+You’ll notice that two new directories have appeared in your repository, containing the files that are in the original RedisModulesSDK and hiredis repositories.
+
+Before we continue, we need to commit these submodules. If you run git status, you’ll see that GIt is aware that we’ve added two submodules (they are tracked in a special file called `.gitsubmodules`) but they have not yet been committed:
```
[the remark this answers is not in the document]
Comment (borjasotomayor) [reply website/content/labs/lab8.md]: Fixed in fe72c9b

File: website/content/labs/lab8.md [hunk 5]
```diff
@@ -0,0 +1,376 @@
...
+
+You will now be shown the pipeline's configuration (you will also be able to access this page through your dashboard, which will now include a `CNETID-pipeline` pipeline). Notice how there is no production application in your pipeline. In the real world, this would mean that your end users don’t see anything!!!
+
+##  Task 2: Add a production application
+
+In this task, you’ll add a production application to your pipeline. Unlike the app in staging, your production app won't be associated with any specific repo. Instead, once your staging application is good to go, you can *promote* it to be the production application. That means the production application will just take whatever is in the staging app, and will run that in production. Later in the lab you may see Heroku messages that talk about "slugs"; in Heroku parlance, a "slug" is a pre-compiled version of your app that is ready to run. So, when we promote to production, we don't actually rebuild the application; we just take the "slug" that is in staging, and copy it to production.
```
[the remark this answers is not in the document]
Comment (borjasotomayor) [reply website/content/labs/lab8.md]: Fixed in fe72c9b

File: website/content/labs/lab8.md (hunk 1 above)
Comment (borjasotomayor) [reply website/content/labs/lab8.md]: Fixed in fe72c9b

**Pull request merged**